src/include/grpc_transcoding/path_matcher.h

```diff
@@ -281,7 +294,8 @@ void ExtractBindingsFromPath(const std::vector<HttpTemplate::Variable>& vars,
     // Joins parts with "/"  to form a path string.
     for (size_t i = var.start_segment; i < end_segment; ++i) {
       // For multipart matches only unescape non-reserved characters.
-      binding.value += UrlUnescapeString(parts[i], !is_multipart);
+      binding.value += UrlUnescapeString(
+          parts[i], fully_decode_reserved_expansion || !is_multipart);
```

Comment:
Hmm,   we should not decode %2F even with fully_decode_reserved_expansion_

Reply:
Hm, not decoding everything beats the purpose of this change..

Reply:
Depends on how we set this flag fully_decode_reserved_expansion_?   If it is from option.http,  I think we should follow it.
At least its comment clearly say so.

We can add a flag in envoy grpc_transcode filter_config, as always_decode_reserved_.   it may serve your purpose. I will go for it.

Reply:
Since you are here,  maybe we can support both:
1) one from proto file option.http,  and 2) one from grpc_trancode filter_config.

since they are mutually exclusive,  we can do: if 2) exists, don't use 1), otherwise check 1).

Reply:
Sounds good, let me rename the function/field with corresponding docs.

Reply:
Works for me

Reply:
Added support for both options, covered logic of all their combinations by tests